src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesVsPatterns.cs

```diff
@@ -2071,5 +2071,28 @@ static void M1(C c, C c2)
                 Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "c.c.c").WithLocation(19, 17)
                 );
         }
+
+        [Fact]
+        [WorkItem(40629, "https://github.com/dotnet/roslyn/issues/40629")]
+        public void NullTestInSwitch_01()
+        {
+            CSharpCompilation c = CreateNullableCompilation(@"
+#nullable enable
+class C
+{
+    void M(object? p)
+    {
+        switch (p)
```

Comment:
Is it expected that this change would also fix a case like the following?

```cs
switch (p == null)
{
    case true:
        return;
}

p.ToString();
```

Reply:
I would not expect that to work any more than I would expect this to work:
`if ((p == null) == true) return;`

Reply:
Makes sense, thanks.